src/main/java/com/marklogic/grove/boot/auth/AuthController.java

```diff
@@ -35,11 +38,29 @@
 	public SessionStatus login(@RequestBody LoginRequest request, HttpSession session, HttpServletResponse response) {
 		logger.info("Logging in user: " + request.getUsername());
 		// TODO Handle error appropriately
-		DatabaseClient client = DatabaseClientFactory.newClient(
-			markLogicConfig.getHost(),
-			markLogicConfig.getRestPort(),
-			new DatabaseClientFactory.DigestAuthContext(request.getUsername(), request.getPassword())
-		);
+
+		DatabaseClient client;
+
+		// if connecting to MarkLogic using SSL
+		// DHS only uses SSL, mlUseSSL in application.properties has to be set to 'true'
+		if (markLogicConfig.getUseSSL()) {
```

Comment:
There's a class in ml-javaclient-util called DefaultConfiguredDatabaseClientFactory that will do all of this for us (ml-gradle and DHF both make heavy use of this class). It depends on an instance of DatabaseClientConfig, which is a bean-like class that defines all the possible properties for constructing a DatabaseClient. 

I'm wondering if MarkLogicConfig could go away in favor of an instance of DatabaseClientConfig. The problem with that is DatabaseClientConfig doesn't have Value annotations on it (nor should it). 

A simple solution is for MarkLogicConfig to exist and then AuthController can construct an instance of DatabaseClientConfig based on the property values in MarkLogicConfig. It will then use an instance of DefaultConfiguredDatabaseClientFactory to construct a DatabaseClient .

Reply:
The details are a bit beyond me, but I do agree that SSL support should not be a Grove-specific thing. There is nothing Grove-specific about the entire config I think, and on that thought I'd agree it would make sense to get rid of MarkLogicConfig.

The Value annotations do look convenient though. Would it make sense to create a wrapper for DbClientConfig, which adds the annotations, maybe add that to app-deployer as well?

Reply:
Hmmm, I'd look at this project perhaps also outside of perspective of (only) Grove. This can be used on its own, on projects where someone will not necessarily use Grove UI. What I'm trying to say - there's nothing in the source code that's Grove-specific, correct?

Reply:
Except the package naming :)..

Reply:
> There's a class in ml-javaclient-util called DefaultConfiguredDatabaseClientFactory that will do all of this for us (ml-gradle and DHF both make heavy use of this class). It depends on an instance of DatabaseClientConfig, which is a bean-like class that defines all the possible properties for constructing a DatabaseClient.
> 
> I'm wondering if MarkLogicConfig could go away in favor of an instance of DatabaseClientConfig. The problem with that is DatabaseClientConfig doesn't have Value annotations on it (nor should it).
> 
> A simple solution is for MarkLogicConfig to exist and then AuthController can construct an instance of DatabaseClientConfig based on the property values in MarkLogicConfig. It will then use an instance of DefaultConfiguredDatabaseClientFactory to construct a DatabaseClient .

@rjrudin, I understand what you're saying. My question: why would we do this?